lib/ex_doc/formatter/html/search_items.ex

```diff
@@ -11,21 +11,22 @@ defmodule ExDoc.Formatter.HTML.SearchItems do
     ["searchNodes=" | SimpleJSON.encode(items)]
   end
 
-  @h2_split_regex ~r/<h2.*?>/
-  @header_body_regex ~r/(?<header>.*)<\/h2>(?<body>.*)/s
+  @h2_split_regex ~r/<h2\b.*?>/
+  @header_body_regex ~r/(?<header>.+)<\/h2>(?<body>.*)/s
   defp extra(%{id: id, title: title, content: content}) do
     [intro | sections] = Regex.split(@h2_split_regex, content)
     intro_json_item = encode("#{id}.html", title, :extras, intro)
 
     section_json_items =
       sections
       |> Enum.map(&Regex.named_captures(@header_body_regex, &1))
+      |> Enum.reject(&is_nil/1)
       |> Enum.map(&extra_section(title, &1["header"], &1["body"], id))
 
     [intro_json_item | section_json_items]
   end
 
-  defp extra_section(title, header, body, id) do
+  defp extra_section(title, header, body, id) when header != nil do
```

Comment:
When would the header be nil?

Reply:
sorry, It was a left over when I was debugging it.

Reply:
removed